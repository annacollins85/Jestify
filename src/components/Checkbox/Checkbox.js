export default {
  name: 'HelloWorld',
  props: {
    checked: {
      type: Boolean,
      default: false
    },
    description: String,
    disabled: {
      type: Boolean,
      default: false
    },
    id: String,
    group: String,
    name: String
  }
  // methods: {
  //   toggl
  // }
}
