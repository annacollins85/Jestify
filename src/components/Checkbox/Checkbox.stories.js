import { storiesOf } from '@storybook/vue'
import { withKnobs, boolean, text } from '@storybook/addon-knobs/vue'
import { withNotes } from '@storybook/addon-notes'

import Checkbox from './Checkbox.vue'

const knobs = (options = {}) => {
  return {
    checked: boolean('checked', options.checked || false),
    description: text('description', options.description || 'lorem ipsum dolor sit amet'),
    disabled: boolean('disabled', options.disabled || false),
    id: text('id', options.id || 'checkbox'),
    group: text('group', options.group || 'group'),
    name: text('name', options.name || 'name')
  }
}

storiesOf('Checkbox', module)
  .addDecorator(withKnobs)
  .add('with radio', withNotes('This a note for the Checkbox component')(() => {
    return ({
      components: { Checkbox },
      template: '<Checkbox v-bind="props" />',
      data () {
        return {
          props: knobs()
        }
      }
    })
  }))
