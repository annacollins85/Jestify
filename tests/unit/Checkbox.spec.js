import { shallow } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'

import Checkbox from '@/components/Checkbox/Checkbox.vue'

describe('Checkbox.vue', () => {
  it('renders the name when passed', () => {
    const name = 'name of Checkbox'
    const wrapper = shallow(Checkbox, {
      propsData: { name }
    })
    expect(wrapper.text()).toMatch(name)
  })
  it('renders the description when passed', () => {
    const description = 'description of Checkbox'
    const wrapper = shallow(Checkbox, {
      propsData: { description }
    })
    expect(wrapper.text()).toMatch(description)
  })
  it('should toggle checked status when clicked', () => {
    const wrapper = shallow(Checkbox)
    wrapper.trigger('click')
    expect(wrapper.props().checked).toBe(true)
    wrapper.trigger('click')
    expect(wrapper.props().checked).toBe(true)
  })
  it('matches snapshot', () => {
    const name = 'Checkbox name'
    const description = 'Checkbox description'
    const renderer = createRenderer()
    const wrapper = shallow(Checkbox, {
      propsData: { name, description }
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
