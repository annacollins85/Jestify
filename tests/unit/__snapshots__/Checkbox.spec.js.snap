// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox.vue matches snapshot 1`] = `
<label class="c-checkbox">
    <input type="checkbox" class="c-checkbox__input c-form__control"> <span class="c-checkbox__indicator"></span> <span class="c-checkbox__label"><h4 class="c-checkbox__name">CheckBox name</h4> <p class="c-checkbox__description">CheckBox description</p></span></label>
`;
